### PR TITLE
initial influxdb auth

### PR DESCRIPTION
**Pull request by benjaminfrank**
I tested the PR on my machine and it was able to connect and write to influxdb using user and password

.

### CODE REVIEW

File: src/rancilio-pid.cpp [hunk 1]
```diff
@@ -1989,6 +1989,11 @@ void setup() {
                 mqtt.setCallback(mqtt_callback);
                 checkMQTT();
             }
+
+            if(INFLUXDB_USER != "" && INFLUXDB_PASSWORD != "") {
```
Comment (endym) [inline src/rancilio-pid.cpp:1993]: String comparisons do not work this way in C/C++. Maybe better use:
`if ((strlen(INFLUXDB_USER) > 0) && (strlen(INFLUXDB_PASSWORD) > 0))`
or maybe safer (didn't tested this code):
`if ((String(INFLUXDB_USER).length() > 0) && (String(INFLUXDB_PASSWORD).length() > 0))`

Comment (benjaminfrank) [reply src/rancilio-pid.cpp]: done

**Review by endym — COMMENTED** [review]
Looks good! Did you tested it with non-empty strings for `INFLUXDB_USER` and `INFLUXDB_PASSWORD`?

**Comment by benjaminfrank** [comment]
yes, that is what I am running on my machine

**Review by endym — APPROVED** [review]
Ok, then it's ok from my side :-)

**Comment by benjaminfrank** [comment]
hm it might be a good idea to check if it properly works with an empty string though

**Comment by endym** [comment]
Yes, indeed.

**Comment by fiendie** [comment]
@benjaminfrank I cherry-picked your commit into #218 and added API Token authentication for InfluxDB >=2.x so this PR can be closed

**Pull request closed without merging**